.github/workflows/github.yml

```
@@ -312,6 +375,92 @@ jobs:
         DEPLOY_RSA: ${{ secrets.DEPLOY_RSA }}
         PACKAGE_EXTENSION: ${{ matrix.extension }}
 
+    - name: Prepare partial JSON with build informations
+      id: make_partial_json
+      shell: bash
+      run: |
+        # Decide cache tool and family
+        if [[ "${{ startsWith(matrix.platform, 'msvc') }}" == "true" ]]; then
+          TOOL="sccache";  CMD="sccache --show-stats"; FAMILY="windows-msvc"
+        elif [[ "${{ startsWith(matrix.platform, 'mingw_') }}" == "true" ]]; then
+          TOOL="ccache";   CMD="ccache -s";            FAMILY="windows-mingw"
+        elif [[ "${{ startsWith(matrix.platform, 'mac') }}" == "true" ]]; then
+          TOOL="ccache";   CMD="ccache -s";            FAMILY="macos"
+        elif [[ "${{ matrix.platform }}" == "ios" ]]; then
+          TOOL="ccache";   CMD="ccache -s";            FAMILY="ios"
+        elif [[ "${{ startsWith(matrix.platform, 'android') }}" == "true" ]]; then
+          TOOL="ccache";   CMD="ccache -s";            FAMILY="android"
+        else
+          TOOL="ccache";   CMD="ccache -s";            FAMILY="other"
+        fi
+        
+        # Raw stats (never fail)
+        STATS="$($CMD || true)"
+        
+        # Parse hits/misses
+        if [[ "$TOOL" == "sccache" ]]; then
+          HITS="$(printf '%s\n' "$STATS" | awk '/^Cache hits/   {print $NF; exit}')"
+          MISS="$(printf '%s\n' "$STATS" | awk '/^Cache misses/ {print $NF; exit}')"
+        else
+          HITS="$(printf '%s\n' "$STATS" | awk '/^[[:space:]]*Hits:/   {print $2; exit}')"
+          MISS="$(printf '%s\n' "$STATS" | awk '/^[[:space:]]*Misses:/ {print $2; exit}')"
+        fi
+        [[ "$HITS" =~ ^[0-9]+$ ]] || HITS=0
+        [[ "$MISS" =~ ^[0-9]+$ ]] || MISS=0
+        TOTAL=$((HITS+MISS))
+        # Two-decimal percent, e.g. 99.36%
+        RATE=$(awk -v h="$HITS" -v t="$TOTAL" 'BEGIN{ if(t>0) printf("%.2f%%",(100.0*h)/t); else print "n/a" }')
```

looks quite monstrous to me, can be replaced with a much easier to understand construct:
```suggestion
        if (( $TOTAL > 0 )) ; then
          percentage=$(bc -e "scale=2; (100 * $HITS) / $TOTAL")
          RATE="${percentage}%"
        else
          RATE="n/a"
        fi
```